tt_eager/dtx/CMakeLists.txt

```diff
@@ -0,0 +1,37 @@
+
+set(DTX_SRCS
+    ${CMAKE_CURRENT_SOURCE_DIR}/dtx.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/dtx_evaluate.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/parallelize_generic_tensor_slice.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/pass_collapse_transformations.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/pass_convert_tensor_layout_3d_conv_act_to_2Dmatrix.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/pass_generate_transfer_addresses.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/pass_reverse_transformations.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/pass_transpose_xy.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/pass_transpose_yz.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/pass_row_major_memory_store.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/pass_row_major_memory_store_blocks.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/pass_convert_abstract_tensor_to_channels_last_layout.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/pass_tilize.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/pass_pad_2d_matrix.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/pass_block_2d_matrix.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/pass_block_2d_with_duplicate_blocks.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/pass_generate_groups_outermost_dim.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/pass_util.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/pass_simple_high_level_pass.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/pass_conv_transform.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/util.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/util_vector_of_ints.cpp
+)
+
+add_library(dtx STATIC ${DTX_SRCS})
```

Comment:
Nice to see `object` lib used 👍 

It also seem that you are aware that it does not cary transitive dependencies, so if a lib links something else - that link and other target params do not propagate to the consumer. 

Is that why you use STATIC here?